src/main/java/io/opentracing/contrib/jdbi/OpenTracingCollector.java

```diff
@@ -0,0 +1,151 @@
+        Span parent = (Span)statementContext.getAttribute(PARENT_SPAN_ATTRIBUTE_KEY);
+        if (parent == null && this.activeSpanSource != null) {
+            parent = this.activeSpanSource.activeSpan(statementContext);
+        }
+        if (parent != null) {
+            builder = builder.asChildOf(parent);
+        }
+        Span collectSpan = builder.start();
+        try {
+            collectSpan.log("Raw SQL", statementContext.getRawSql());
+        } finally {
+            collectSpan.finish(nowMicros);
+        }
+    }
+
+    /**
+     * Establish an explicit parent relationship for the (child) Span associated with a SQLStatement.
+     *
+     * @param statement the JDBI SQLStatement which will act as the child of `parent`
+     * @param parent the parent Span for `statement`
+     */
+    public static void setParent(SQLStatement<?> statement, Span parent) {
+        statement.getContext().setAttribute(PARENT_SPAN_ATTRIBUTE_KEY, parent);
+    }
+
+    /**
+     * OperationNamer allows the OpenTracingCollector user to control the precise naming of OpenTracing Spans emitted
+     * by the collector.
+     *
+     * @see OpenTracingCollector#OpenTracingCollector(Tracer, OperationNamer)
+     */
+    public interface OperationNamer {
+        public static OperationNamer DEFAULT = new OperationNamer() {
+            @Override
+            public String generateOperationName(StatementContext ctx) {
+                return "DBI Statement";
+            }
+        };
+
+        /**
+         * Transform an DBI StatementContext into an OpenTracing Span operation name.
+         *
+         * @return an operation name suitable for an OpenTracing Span
+         */
+        public String generateOperationName(StatementContext ctx);
+    }
+
+    /**
+     * An abstract API that allows the OpenTracingCollector to customize how parent Spans are discovered.
+     *
+     * For instance, if Spans are stored in a thread-local variable, an ActiveSpanSource could access them like so:
+     * <p>Example usage:
+     * <pre>{@code
+     * public class SomeClass {
+     *     // Thread local variable containing each thread's ID
+     *     private static final ThreadLocal<Span> activeSpan =
+     *         new ThreadLocal<Span>() {
+     *             @Override protected Integer initialValue() {
+     *                 return null;
+     *             }
+     *         };
+     *     };
+     *
+     *     ... elsewhere ...
+     *     {
+     *         ActiveSpanSource spanSource = new ActiveSpanSource() {
+     *             @Override
+     *             public Span activeSpan(StatementContext ctx) {
+     *                 // (In this example we ignore `ctx` entirely)
+     *                 return activeSpan.get();
+     *             }
+     *         };
+     *         OpenTracingCollector otColl = new OpenTracingCollector(tracer, spanSource);
+     *         ...
+     *     }
+     * }</pre>
+     */
+    public interface ActiveSpanSource {
```

Comment:
Yeah I think it looks good! So basically, they can either set the parent span manually, or otherwise implement a method that returns the current span (which then runs automatically)? It definitely seems easier if they're doing lots of calls than to have to set the parent span manually every time